ecs-cli/modules/utils/compose/ecs_params_reader_test.go

```diff
@@ -512,6 +512,65 @@ task_definition:
 	}
 }
 
+func TestReadECSParams_WithServiceDiscoveryAllFields(t *testing.T) {
+	ecsParamsString := `version: 1
+run_params:
+  service_discovery:
+    container_name: nginx
+    container_port: 80
+    private_dns_namespace:
+      vpc: vpc-8BAADF00D
+      id: ns-CA15CA15CA15CA15
+      name: corp
+      description: This is a private namespace
+    public_dns_namespace:
+      id: ns-C0VF3F3
+      name: amazon.com
+    service_discovery_service:
+      name: mysds
+      description: This is an SDS
+      dns_config:
+        type: A
+        ttl: 60
+      routing_policy: MULTIVALUE
+      healthcheck_custom_config:
+        failure_threshold: 1`
+
+	content := []byte(ecsParamsString)
+
+	tmpfile, err := ioutil.TempFile("", "ecs-params")
+	assert.NoError(t, err, "Could not create ecs-params tempfile")
+
+	ecsParamsFileName := tmpfile.Name()
+	defer os.Remove(ecsParamsFileName)
+
+	_, err = tmpfile.Write(content)
+	assert.NoError(t, err, "Could not write data to ecs-params tempfile")
+
+	err = tmpfile.Close()
+	assert.NoError(t, err, "Could not close tempfile")
+
+	ecsParams, err := ReadECSParams(ecsParamsFileName)
+
+	if assert.NoError(t, err) {
+		serviceDiscovery := ecsParams.RunParams.ServiceDiscovery
+		assert.Equal(t, "nginx", serviceDiscovery.ContainerName, "Expected ContainerName to match")
+		assert.Equal(t, aws.Int64(80), serviceDiscovery.ContainerPort, "Expected ContainerPort to match")
+		assert.Equal(t, "vpc-8BAADF00D", serviceDiscovery.PrivateDNSNamespace.VPC, "Expected VPC to match")
+		assert.Equal(t, "ns-CA15CA15CA15CA15", serviceDiscovery.PrivateDNSNamespace.ID, "Expected private namespace ID to match")
+		assert.Equal(t, "corp", serviceDiscovery.PrivateDNSNamespace.Name, "Expected private namespace Name to match")
+		assert.Equal(t, "This is a private namespace", serviceDiscovery.PrivateDNSNamespace.Description, "Expected private namespace description to match")
+		assert.Equal(t, "ns-C0VF3F3", serviceDiscovery.PublicDNSNamespace.ID, "Expected public namespace ID to match")
+		assert.Equal(t, "amazon.com", serviceDiscovery.PublicDNSNamespace.Name, "Expected public namespace name to match")
+		assert.Equal(t, "mysds", serviceDiscovery.ServiceDiscoveryService.Name, "Expected SDS Name to match")
```

Comment:
Just for readability, any chance we can extract `serviceDiscovery.ServiceDiscoveryService` into a local var within this block and call the other fields on it?

Reply:
Sure